otx/algorithms/segmentation/adapters/mmseg/models/losses/detcon_loss.py

```diff
@@ -42,11 +42,11 @@ def __init__(
 
     def get_distributed_tensors(self, target1, target2, batch_size, num_samples, num_features, device):
         """Grab tensors across replicas during distributed training."""
-        num_gpus = torch.cuda.device_count()
-        if num_gpus > 1 and torch.distributed.is_initialized() and self.use_replicator_loss:
+        if torch.distributed.is_initialized() and self.use_replicator_loss:
```

Comment:
Is `torch.distributed.is_initialized()` True when only using multi gpu?
And during multi gpu, do `torch.cuda.device_count()` and `dist.get_world_size()` have the same value?
If so, I agree with this change.

Reply:
1) Is torch.distributed.is_initialized() True when only using multi gpu? => yes
2) do torch.cuda.device_count() and dist.get_world_size() have the same value? => No. Although machine has 4 GPUs user can only use 2 GPUs for training. In that case, torch.cuda.device_count() is 4 and dist.get_world_size() is 2.

Reply:
> 1. Is torch.distributed.is_initialized() True when only using multi gpu? => yes
> 2. do torch.cuda.device_count() and dist.get_world_size() have the same value? => No. Although machine has 4 GPUs user can only use 2 GPUs for training. In that case, torch.cuda.device_count() is 4 and dist.get_world_size() is 2.

Oh, if so, previous implementation was bug. Thanks for finding.
I left some more comments.